src/core/thread/dua_manager.cpp

```diff
@@ -33,51 +33,114 @@
+    , mRegisterCurrentChildIndex(false)
+#endif
 {
+    mDelay.mValue = 0;
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     mDomainUnicastAddress.Clear();
     mDomainUnicastAddress.mAddressOrigin      = OT_ADDRESS_ORIGIN_THREAD;
-    mDomainUnicastAddress.mPreferred          = true;
+    mDomainUnicastAddress.mPreferred          = false;
     mDomainUnicastAddress.mValid              = true;
     mDomainUnicastAddress.mScopeOverride      = Ip6::Address::kGlobalScope;
     mDomainUnicastAddress.mScopeOverrideValid = true;
 
     mFixedDuaInterfaceIdentifier.Clear();
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    mChildDuaMask.Clear();
+    mChildDuaRegisterMask.Clear();
+#endif
+
+    Get<Coap::Coap>().AddResource(mDuaNotification);
 }
 
-void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefixState aState)
+void DuaManager::HandleDomainPrefixUpdate(BackboneRouter::Leader::DomainPrefixState aState)
 {
-    const otIp6Prefix *prefix;
+    const otIp6Prefix *prefix = nullptr;
+
+    if (aState == BackboneRouter::Leader::kDomainPrefixRefreshed)
+    {
+        if (mIsDuaPending)
+        {
+            IgnoreError(Get<Coap::Coap>().AbortTransaction(&DuaManager::HandleDuaResponse, this));
+        }
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+        // Remove Children's DUAs
+        for (Child &child : Get<ChildTable>().Iterate(Child::kInStateValid))
+        {
+            uint16_t childIndex = Get<ChildTable>().GetChildIndex(child);
+
+            if (mChildDuaMask.Get(childIndex))
+            {
+                child.RemovePreviousDomainUnicastAddress();
+            }
+        }
+
+        mChildDuaMask.Clear();
+        mChildDuaRegisterMask.Clear();
+        mRegisterCurrentChildIndex = false;
+#endif
+    }
+
+    if ((aState == BackboneRouter::Leader::kDomainPrefixAdded) ||
+        (aState == BackboneRouter::Leader::kDomainPrefixRefreshed) ||
+        (aState == BackboneRouter::Leader::kDomainPrefixUnchanged))
+    {
+        prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
 
+        OT_ASSERT(prefix != nullptr);
+    }
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
     if ((aState == BackboneRouter::Leader::kDomainPrefixRemoved) ||
         (aState == BackboneRouter::Leader::kDomainPrefixRefreshed))
     {
-        Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
+        RemoveDomainUnicastAddress();
     }
 
-    VerifyOrExit((aState == BackboneRouter::Leader::kDomainPrefixAdded) ||
-                     (aState == BackboneRouter::Leader::kDomainPrefixRefreshed),
+    VerifyOrExit(aState == BackboneRouter::Leader::kDomainPrefixAdded ||
+                     aState == BackboneRouter::Leader::kDomainPrefixRefreshed ||
+                     (aState == BackboneRouter::Leader::kDomainPrefixUnchanged &&
+                      !Get<ThreadNetif>().HasUnicastAddress(
+                          GetDomainUnicastAddress())) /* in case removed for some reason e.g. the kDuaInvalid response
```

Comment:
It is not clearly defined in specification, below is personal understanding ,fyi

`ST_DUA_INVALID Registration rejected: Target EID is not a valid DUA; ; e.g. incorrect Domain Prefix.`

Since the DUA registered is responded with incorrect Domain Prefix, it may not make sense to re-generate a new DUA with the incorrect Domain Prefix immediately and register again.

Generally the device should only re-generate new DUA when new Domain Prefix is available. And here also processes 'aState == BackboneRouter::Leader::kDomainPrefixUnchanged && !Get<ThreadNetif().HasUnicastAddress(GetDomainUnicastAddress()' to make it robust for the unexpected `ST_DUA_INVALID` response from PBBR, like the intentional behavior during certification.